### ISSUE TITLE
[feature]: Collection with variable

### ISSUE THREAD

**Issue by shibilpk** [hoppscotch/hoppscotch#3736]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

We need set variable for collection itselfto 

### Why should this be worked on?

If working on more projects. Sometimes its easy to manage variables if there option to set variables for collection. In that case, we need to create less environment.

**Comment by GauthierPLM** [comment]
My team has around 30 to 40 collections in our team space, with one collection for each micro service. As we have many variables that are specific to a given collection, it does not make sense for us to put them in the environment, as the number of variables would grow to an unmanageable number.

Having collection-level variables like what Postman offers would allow us to properly manage our collections / variables with the granularity we need. This is the last blocker we identified to switch from Postman to Hoppscotch.

**Comment by liyasthomas** [comment]
Thanks for your feedback, we'll look into adding this to the roadmap.

**Comment by GauthierPLM** [comment]
@liyasthomas is there any place to see the roadmap / priority you'd give to this feature?

**Comment by weinull** [comment]
looking forward to this feature

**Comment by m-amiri** [comment]
same issue

**Comment by BoscoDomingo** [comment]
I echo the rest. It's the missing bit to be able to replace Postman

**Comment by ankitgirase** [comment]
@liyasthomas can you please share progress for this feature. We are making transition from postman to hoppscotch and only this one feature is limiting us, your response will be valuable for us.

**Comment by ankit-girase-zw** [comment]
@GauthierPLM did your team shifted to hoppscotch from postman ? If yes how did you managed collection variables ?

**Comment by GauthierPLM** [comment]
@ankit-girase-zw because this feature was central in our workflow, we decided to stick with Postman unfortunately.

**Comment by lsdgem** [comment]
I strongly agree with all the other people in this thread. This is genuinely the only reason we can not switch from Postman to Hoppscotch

**Comment by Aurgil** [comment]
We need this feature too to use Hoppscotch.

**Comment by carterclark** [comment]
Would love for this to be added

**Comment by ygg-shibil** [comment]
@liyasthomas Any Update on this Feature? Please give an update

**Comment by m-amiri** [comment]
any updates? 👀

**Comment by faritor** [comment]
Hi,@liyasthomas ,

in which version will this rough plan be implemented, or is there an expected time? this is one of the key points that prevent me from migrating as a whole

**Comment by mcules** [comment]
This is the feature I miss the most. It would make perfect sense to be able to store variables under collections or even directly within individual folders. I use this functionality extensively in Postman and hope it will be available here soon.
At the moment, this is the only thing keeping me from fully switching to Hoppscotch.

**Comment by mashmex** [comment]
Hi together,

This feature is very important, especially when migrating from other services such as Postman. Since you don't always create separate workspaces for a project/customer. In Postman we were able to set variables on collections there, it would be important for this feature to be available in Hoppscotch as soon as possible.

Thank you.

**Comment by liyasthomas** [comment]
We’ve begun working on this feature request. We apologize for the delay. We expect this feature to be shipped soon.

**Comment by jamesgeorge007** [comment]
Hi, the [v2025.8.0](https://github.com/hoppscotch/hoppscotch/releases/tag/2025.8.0) release is now live and includes support for collection variables, closing this issue. Please feel free to reach out if you have any questions.
